tests/test_patron.py

```diff
@@ -0,0 +1,308 @@
+        assert next(patron_xml_records) == student_patron_record_2
+
+
+def test_populate_patron_common_fields_staff_all_values_success(
+    staff_patron_template, staff_patron_all_values_dict
+):
+    patron_xml_record = populate_patron_common_fields(
+        staff_patron_template,
+        staff_patron_all_values_dict,
+        SIX_MONTHS,
+        TWO_YEARS,
+    )
+    assert patron_xml_record.primary_id.string == "STAFF_KRB_NAME@MIT.EDU"
+    assert patron_xml_record.expiry_date.string == "2023-09-01Z"
+    assert patron_xml_record.purge_date.string == "2025-09-01Z"
+    assert (
+        patron_xml_record.emails.email.email_address.string == "STAFF_KRB_NAME@MIT.EDU"
+    )
+    assert (
+        patron_xml_record.user_identifiers.user_identifier.value.string == "222222222"
+    )
+    assert (
+        patron_xml_record.find_all("user_identifier")[1].value.string
+        == "22222222222222"
+    )
+
+
+def test_populate_patron_common_fields_staff_no_krb_name_but_email_success(
+    staff_patron_template, staff_patron_all_values_dict
+):
+    staff_patron_all_values_dict["KRB_NAME_UPPERCASE"] = None
+    patron_xml_record = populate_patron_common_fields(
+        staff_patron_template,
+        staff_patron_all_values_dict,
+        SIX_MONTHS,
+        TWO_YEARS,
+    )
+    assert patron_xml_record.primary_id.string == "STAFF_KRB_NAME@MIT.EDU"
+    assert patron_xml_record.expiry_date.string == "2023-09-01Z"
+    assert patron_xml_record.purge_date.string == "2025-09-01Z"
+    assert (
+        patron_xml_record.emails.email.email_address.string == "STAFF_KRB_NAME@MIT.EDU"
+    )
+    assert (
+        patron_xml_record.user_identifiers.user_identifier.value.string == "222222222"
+    )
+    assert (
+        patron_xml_record.find_all("user_identifier")[1].value.string
+        == "22222222222222"
+    )
+
+
+def test_populate_patron_common_fields_staff_null_values_success(
+    staff_patron_template,
+):
+    staff_patron_null_values_dict = {
+        "FULL_NAME": None,
+        "OFFICE_ADDRESS": None,
+        "OFFICE_PHONE": None,
+        "MIT_ID": "222222222",
+        "EMAIL_ADDRESS": None,
+        "APPOINTMENT_END_DATE": None,
+        "KRB_NAME_UPPERCASE": None,
+        "LIBRARY_PERSON_TYPE_CODE": None,
+        "LIBRARY_PERSON_TYPE": None,
+        "ORG_UNIT_ID": None,
+        "ORG_UNIT_TITLE": None,
+        "POSITION_TITLE": None,
+        "DIRECTORY_TITLE": None,
+        "LIBRARY_ID": None,
+    }
+    patron_xml_record = populate_patron_common_fields(
+        staff_patron_template,
+        staff_patron_null_values_dict,
+        SIX_MONTHS,
+        TWO_YEARS,
+    )
+    assert patron_xml_record.expiry_date.string == "2023-09-01Z"
+    assert patron_xml_record.purge_date.string == "2025-09-01Z"
+    assert (
+        patron_xml_record.user_identifiers.user_identifier.value.string == "222222222"
+    )
+    assert not list(list(patron_xml_record.emails.children))
```

Comment:
Is there a reason for the double `list` here and in the corollary student test below?

Reply:
Find & replace gone wrong that I missed